pydash/numerical.py

```diff
@@ -383,6 +386,43 @@ def std_deviation(array):
     return math.sqrt(variance(array))
 
 
+def call_math_operator(value1, value2, callback):
+    """Return the result of the math operation on the given values.
+
+    Args:
+        value1 (int/float/str): Value passed by the user.
+        value2 (int/float/str): Value passed by the user.
+        callback (mixed, optional): Callback applied on the given values.
+
+    Returns:
+        float: Result after the math operation is conducted on the given
+            values.
+
+    Example:
+
+        >>> call_math_operator(5, 4, lambda x, y: x - y)
+        1.0
+        >>> call_math_operator(4, 6, lambda x, y: x * y)
+        24.0
+        >>> call_math_operator('5', '10', lambda x, y: x + y)
+        15.0
+        >>> call_math_operator('25', '5', lambda x, y: x / y)
+        5.0
+
+    .. versionadded:: TODO
+    """
+    # If the user has given any Nonetype values, we assign 0 as the default
+    # value to the values.
+    value1 = 0 if not value1 else value1
+    value2 = 0 if not value2 else value2
+
+    if isinstance(value1, str) or isinstance(value2, str):
```

Comment:
This comparison won't cover Python 2 since it has the unicode string type. Need to use the `string_types` from `_compat.py`.

However, we should probably instead check if the values are numbers and if not then try to convert them to float since the user may use non-strings or non-numbers that could support conversion to float.

This will tie into whatever policy we decide to implement for handling of non-convertible types. Generally, I think it better if we raise an exception if a non-convertible object is passed in. With that in mind, I think we should do the following:

- If value is `None`, convert to a default value.
- If value is `not pyd.is_number(val)`, then try to convert it to a float but `pass` on any exception.
- If float conversion fails, go ahead with operation and let it throw it's exception (i.e. we don't need to throw it explicitly). This way the exception occurs during the operation and not the attempted float conversion.